### ISSUE TITLE
Make m:n nested updates API consistent with 1:1 and 1:m

### ISSUE THREAD

**Issue by dmarcelino** [balderdashy/waterline#965]
Nested updates in the form of `User.update({id: 1 }, { posts: [ { id: 1, title: 'Test post - updated' }] })` work for 1:1 and 1:m associations but breaks for m:n associations. The API for this should be consistent between association types.

PR balderdashy/waterline-adapter-tests#51 provides the failing tests for this feature.

**#292 - Add `.add()`, `.remove()`, `.save()` support for 1-1 and 1-* associations** complements this by making the API across 1:1, 1:m and m:n the same for all operations.

**Comment by devinivy** [comment]
I surprised this is an issue, honestly.  What occurs instead of a successful update?  Is it possibly updated in the database, but not reflected in the results waterline returns?

**Comment by dmarcelino** [comment]
The child values are left untouched but the association to the child is set correctly.

The tests do a `.find()` after the update (https://github.com/balderdashy/waterline-adapter-tests/pull/51/files#diff-4a8c2e665523976121614a83468c4a69L176) so I don't think it was updated in the DB.

**Comment by devinivy** [comment]
We'll wanna look around here: https://github.com/balderdashy/waterline/blob/master/lib/waterline/utils/nestedOperations/update.js#L167-L169

**Comment by dmarcelino** [comment]
@devinivy you are like a travel guide to waterline core code! :smiley:

**Comment by sailsbot** [comment]
Thanks for posting, @dmarcelino.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by devinivy** [comment]
This is the sole blocker for https://github.com/balderdashy/waterline-adapter-tests/pull/51

**Comment by sailsbot** [comment]
Thanks for posting, @dmarcelino.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by sailsbot** [comment]
Thanks for posting, @dmarcelino.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
